ambry-router/src/main/java/com.github.ambry.router/GetBlobOperation.java

```diff
@@ -409,12 +429,14 @@ private void updateChunkingAndSizeMetricsOnSuccessfulGet() {
   private class GetChunk {
     // the operation tracker used to track the operation on the current chunk.
     private OperationTracker chunkOperationTracker;
-    // whether the operation on the current chunk has completed.
-    private boolean chunkCompleted;
     // the blob id of the current chunk.
     private BlobId chunkBlobId;
+    // whether the operation on the current chunk has completed.
+    private boolean chunkCompleted;
+    // indicates that the set chunk exception should not be overwritten even when the operation tracker reports success.
```

Comment:
You could prepend this by saying "in general, when the operation tracker returns success, any previously saved exceptions are cleared. This flag indicates that ..."